### PR TITLE
migrate webhooks

### LINKED ISSUE #1738: Remove laravel-bitflags // rewrite webhook events
Currently the webhooks sends requests which are saved in bitflags via the package [uwumarie/laravel-bitflags](https://github.com/NyCodeGHG/laravel-bitflags). This package is not maintained and causes issues to our upgrading workflow.

We should remove the dependency and maybe just save the choosen events in another table.

image

### LINKED ISSUE #1744: Http 500 error on webhook creation
Report: https://discord.com/channels/906993329777020928/1129291909760823408/1129334281991372830

```
[2023-07-14 10:49:55] production.ERROR: App\Http\Controllers\Backend\Auth\AccessTokenController::extendResponseWithWebhookData(): Return value must be of type Nyholm\Psr7\Response, Illuminate\Http\Response returned {"exception":"[object] (TypeError(code: 0): App\\Http\\Controllers\\Backend\\Auth\\AccessTokenController::extendResponseWithWebhookData(): Return value must be of type Nyholm\\Psr7\\Response, Illuminate\\Http\\Response returned at app/Http/Controllers/Backend/Auth/AccessTokenController.php:55)
[stacktrace]
#0 app/Http/Controllers/Backend/Auth/AccessTokenController.php(20): App\\Http\\Controllers\\Backend\\Auth\\AccessTokenController->extendResponseWithWebhookData()
#1 vendor/laravel/passport/src/Http/Controllers/HandlesOAuthErrors.php(24): App\\Http\\Controllers\\Backend\\Auth\\AccessTokenController->App\\Http\\Controllers\\Backend\\Auth\\{closure}()
#2 app/Http/Controllers/Backend/Auth/AccessTokenController.php(19): Laravel\\Passport\\Http\\Controllers\\AccessTokenController->withErrorHandling()
#3 vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(46): App\\Http\\Controllers\\Backend\\Auth\\AccessTokenController->issueToken()
#4 vendor/laravel/framework/src/Illuminate/Routing/Route.php(259): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#5 vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#6 vendor/laravel/framework/src/Illuminate/Routing/Router.php(798): Illuminate\\Routing\\Route->run()
#7 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#8 vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php(126): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#9 vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php(57): Illuminate\\Routing\\Middleware\\ThrottleRequests->handleRequest()
#10 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Routing\\Middleware\\ThrottleRequests->handle()
#11 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#12 vendor/laravel/framework/src/Illuminate/Routing/Router.php(797): Illuminate\\Pipeline\\Pipeline->then()
#13 vendor/laravel/framework/src/Illuminate/Routing/Router.php(776): Illuminate\\Routing\\Router->runRouteWithinStack()
#14 vendor/laravel/framework/src/Illuminate/Routing/Router.php(740): Illuminate\\Routing\\Router->runRoute()
#15 vendor/laravel/framework/src/Illuminate/Routing/Router.php(729): Illuminate\\Routing\\Router->dispatchToRoute()
#16 vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(190): Illuminate\\Routing\\Router->dispatch()
#17 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#18 vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(59): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#19 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle()
#20 vendor/fruitcake/laravel-cors/src/HandleCors.php(52): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#21 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Fruitcake\\Cors\\HandleCors->handle()
#22 vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#23 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle()
#24 vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#25 vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#26 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#27 vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#28 vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#29 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#30 vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#31 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#32 vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#33 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#34 vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#35 vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\\Pipeline\\Pipeline->then()
#36 vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(134): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#37 public/index.php(61): Illuminate\\Foundation\\Http\\Kernel->handle()
#38 {main}
"}
```

### PR DESCRIPTION
**Pull request by NyCodeGHG**
This pr migrates webhooks from the bitflags system to a new table.
Existing data will be migrated, but especially that code should be tested locally and reviewed before pushing it into prod.

This pr also fixes #1744 and #1738

I'm also not sure about the migrations I've written, about the reversibility in particular.

Nothing about the current webhook flow gets changed here, but I suggest we also simplify that later.

### CODE REVIEW

**Comment by NyCodeGHG** [comment]
I don't know why the ci fails, locally the package installation works fine for me

File: database/migrations/2023_08_19_161454_migrate_webhook_events_from_bitmasks_to_table.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const CHECKIN_DELETE = 1 << 2;
+    const NOTIFICATION = 1 << 3;
+
+    public function up(): void {
+        DB::table('webhooks')->orderBy('id')->chunkById(100, function (Collection $webhooks) {
+            foreach ($webhooks as $webhook) {
+                $id = $webhook->id;
+                $events_bits = $webhook->events;
+                if ($events_bits & self::CHECKIN_CREATE) {
+                    self::insertWebhookEvent($id, 'CHECKIN_CREATE');
```
Comment (HerrLevin) [inline database/migrations/2023_08_19_161454_migrate_webhook_events_from_bitmasks_to_table.php]: Ist der Enum nicht case sensitive? Wäre dann nicht `'checkin_create'` sinnvoller oder denke ich gerade falsch?

File: database/migrations/2023_08_19_174051_drop_webhook_events_column.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+return new class extends Migration
+{
+    public function up(): void {
+        Schema::table('webhooks', static function (Blueprint $table) {
+            $table->dropColumn('events');
+        });
+    }
+
+    public function down(): void {
+    }
```
Comment (HerrLevin) [inline database/migrations/2023_08_19_174051_drop_webhook_events_column.php]:
```suggestion
    public function down(): void {
            Schema::table('webhooks', function (Blueprint $table) {
            $table->addColumn('uint', 'events');
        });
    }
```

File: database/migrations/2023_08_19_170326_remove_events_column_from_webhooks.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            $table->removeColumn('events');
+        });
+    }
+
+    public function down(): void {
+        Schema::table('webhooks', function (Blueprint $table) {
+            $table->addColumn('uint', 'events');
+        });
+    }
+};
```
Comment (HerrLevin) [inline database/migrations/2023_08_19_170326_remove_events_column_from_webhooks.php]: Shouldn't this be accomplished with database/migrations/2023_08_19_174051_drop_webhook_events_column.php?

**Review by HerrLevin — CHANGES_REQUESTED** [review]

File: database/migrations/2023_08_19_161454_migrate_webhook_events_from_bitmasks_to_table.php (hunk 1 above)
Comment (NyCodeGHG) [reply database/migrations/2023_08_19_161454_migrate_webhook_events_from_bitmasks_to_table.php]: oh ja, hast recht, hatte das enum zuerst uppercase, aber für backwards compat hab ichs dann doch geändert

File: database/migrations/2023_08_19_170326_remove_events_column_from_webhooks.php (hunk 3 above)
Comment (NyCodeGHG) [reply database/migrations/2023_08_19_170326_remove_events_column_from_webhooks.php]: yes, i don't know what i did here, was a bit late when i did that

**Pull request merged**